src/operator/mxnet_op.h

```diff
@@ -441,8 +447,80 @@ struct Kernel<OP, cpu> {
     OP::Map(0, N, args...);
 #endif
   }
+
+  /*!
+   * \brief Launch a tunable OP with explicitly-supplied data type
+   * \tparam DType Data type
+   * \tparam OP type
+   * \tparam Args Varargs type to eventually pass to the OP::Map() functoion
+   * \param s Stream (usually null for CPU)
+   * \param N Number of iterations
+   * \param args Varargs to eventually pass to the OP::Map() functoion
+   * \return Always true
+   */
+  template<typename DType, typename T = OP, typename ...Args>
+  static MSHADOW_CINLINE
+  typename std::enable_if<std::is_base_of<tunable, T>::value, bool>::type
+  LaunchWithType(mshadow::Stream<cpu> *s, const int N, Args... args) {
+    LaunchTuned<T, DType>(s, N, args...);
+    return true;
+  }
+
+  /*!
+   * \brief Launch a tunable OP with implicitly-supplied data type
+   * \tparam DType Data type
+   * \tparam T OP type
+   * \tparam Args Varargs type to eventually pass to the OP::Map() functoion
+   * \param s Stream (usually null for CPU)
+   * \param N Number of iterations
+   * \param args Varargs to eventually pass to the OP::Map() functoion
+   * \return Always true
+   */
+  template<typename DType, typename T = OP, typename ...Args>
+  static MSHADOW_CINLINE
+  typename std::enable_if<std::is_base_of<tunable, T>::value, bool>::type
+  Launch(mshadow::Stream<cpu> *s, const int N, DType *dest, Args... args) {
+    return LaunchWithType<DType, T>(s, N, dest, args...);
```

Comment:
Can we remove LaunchWithType and call LaunchTuned here directly?

Reply:
Yeah, I think I just did it this way so you could tell the difference between the calls. Otherwise it’s really hard to pick out the “dest” argument and understand whats different. I can change though, no problem.

Reply:
Oh yeah, also it acts as an illustration of how to use LaunchByType(), which you’d need to use if DType *dest isn’t the first va_arg argument.
Wdyt?

Reply:
Adding more interface means there will be more backward compatibility issues in the future. 
Let's hold off.

Reply:
ok